pandas/tests/reshape/test_melt.py

```diff
@@ -1218,6 +1218,33 @@ def test_missing_stubname(self, dtype):
         expected.index = expected.index.set_levels(new_level, level=0)
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.parametrize("stubnames", ["year", ["year"]])
```

Comment:
I think it'll be more robust if we test that the function works for any position of year, like `stubnames = ["year", "another stub"]` or `stubnames = ["another stub", "year"]`

I can add more tests once the approach is agreed